libraries/aleth/libp2p/NodeTable.cpp

```diff
@@ -461,6 +461,14 @@ shared_ptr<NodeEntry> NodeTable::handlePong(bi::udp::endpoint const& _from, Disc
   }
 
   m_sentPings.erase(_from);
+  // Remove any other sent pings which was sent to other endpoints for the same nodeID
+  for (auto it = m_sentPings.begin(); it != m_sentPings.end();) {
+    if (nodeValidation.nodeID == it->second.nodeID) {
+      it = m_sentPings.erase(it);
+    } else {
+      ++it;
+    }
+  }
```

Comment:
For instance:
Same node is reported multiple times with different endpoint(different IP and/or port) but same node ID and node is not responding on any of them. For instance case for our consensus nodes which do not open ports. Dropnode will not be called for such nodes if at the time ping expires entry has a different endpoint. There is a chance that such node might never be removed from discovered nodes because of this.

With my change as soon as we receive successful pong all the pings for the same node id are removed from m_sentPings. If within the timeout of 60 seconds, there was no successful response from any endpoint I think it is reasonable to assume that node is not responding.

Can you maybe give me an example what kind of functionality could my change break?